### ISSUE TITLE
Update custom spark build to support s2i-thoth-ubi8-py36

### ISSUE THREAD

**Issue by LaVLaS** [opendatahub-io/s2i-spark-container#6]
With the update to thoth notebook builds, this image no longer imports `pyspark` when built using `s2i-thoth-ubi8-py36` as a base image. I think the fix is to build spark & hadoop from the spark source [repo](https://github.com/apache/spark.git) and then build the wheel file for a local pip install.

**Comment by vpavlin** [comment]
Seems like the thoth base image is not the culprit of the issue - https://github.com/opendatahub-io/s2i-spark-container/commit/9f28593f741a26675bd9b01f3762aabc29782521#diff-bc705d3ab5dd034e3ce0b4943c294895L7 - sorry I missed it in the review

**Comment by LaVLaS** [comment]
I agree. The setup for installing spark with a custom hadoop version needs to be refactored/cleaned up. I *think* we can easily build and install a cleaner pip package from the spark repo that will be automatically discovered by python using the defaults
